doc/project/linking-errors.md

```diff
@@ -0,0 +1,249 @@
+[error] involving instantiated classes:
+[error]   helloworld.HelloWorld$
+[error] There were linking errors
+```
+
+### Hint to recognize this cause
+
+If the `Referring to non-existent` lines mention a library that you depend on, and the first corresponding `called from` lines mention your code, it is very likely that an erroneous `%%` is the cause.
+
+### Solution
+
+Use `%%%` instead of `%%` when depending on other Scala.js librarys:
+
+```scala
+libraryDependencies += "io.suzaku" %%% "boopickle" % "1.3.1"
+```
+
+```
+[info] Fast optimizing .../helloworld/target/scala-2.12/helloworld-fastopt.js
+[success] Total time: 3 s, completed Sep 13, 2019 1:44:25 PM
+```
+
+
+## Using blocking APIs (e.g., `Await.result`)
+
+The JavaScript platform does not support blocking at all.
+Consequently, trying to use block APIs such `Await.result` is not supported, and will result in linking error.
+A typical example was shown in the introduction, and is reproducible with the following code:
+
+```scala
+import scala.concurrent.{Await, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+
+val f = Future { 5 }
+println(Await.result(f, Duration.Inf))
+```
+
+which yields
+
+```
+[error] Referring to non-existent method scala.concurrent.impl.Promise$CompletionLatch.releaseShared(scala.Int)scala.Boolean
+[error]   called from scala.concurrent.impl.Promise$CompletionLatch.apply(scala.util.Try)scala.Unit
+[error]   called from scala.concurrent.impl.Promise$CompletionLatch.apply(java.lang.Object)java.lang.Object
+[error]   called from scala.util.Success.$$anonfun$map$1(scala.Function1)java.lang.Object
+[error]   called from scala.util.Success.map(scala.Function1)scala.util.Try
+[error]   called from scala.concurrent.Future.$$anonfun$map$1(scala.Function1,scala.util.Try)scala.util.Try
+[error]   called from scala.concurrent.impl.Promise$DefaultPromise.$$anonfun$map$1(scala.Function1,scala.util.Try)scala.util.Try
+[error]   called from scala.concurrent.Future.map(scala.Function1,scala.concurrent.ExecutionContext)scala.concurrent.Future
+[error]   called from scala.concurrent.impl.Promise$KeptPromise$Successful.map(scala.Function1,scala.concurrent.ExecutionContext)scala.concurrent.Future
+[error]   called from scala.concurrent.Future$.apply(scala.Function0,scala.concurrent.ExecutionContext)scala.concurrent.Future
+[error]   called from helloworld.HelloWorld$.main()scala.Unit
+[error]   called from core module module initializers
+```
+
+and other errors.
+
+### Hint to recognize this cause
+
+If the `Referring to non-existent` line mentions `CompletionLatch`, it is probably related to trying to block on a `Future`.
+
+### Solution
+
+Always use asynchronous combinators, such as `map`, `filter`, `flatMap`, `foreach`, and/or for comprehensions, when using `Future`.
+Do not use `scala.concurrent.Await`.
+
+
+## Using unsupported JDK libraries (possibly transitively)
```

Comment:
It might be helpful to make an example of a transitive problem (e.g. trying to use `scala.io`)